nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -186,7 +186,7 @@ let
         NIX_EFI_VARS=$(readlink -f "''${NIX_EFI_VARS:-${config.system.name}-efi-vars.fd}")
         # VM needs writable EFI vars
         if ! test -e "$NIX_EFI_VARS"; then
-        ${if cfg.useBootLoader then
+        ${if cfg.useBootLoader && cfg.useDefaultFilesystems then
```

Comment:
Why is it necessary?

Reply:
Because `systemImage` doesn't even eval with a `none` partition layout and an UEFI bootloader enabled.